x-pack/plugins/observability/public/pages/home/index.tsx

```diff
@@ -5,33 +5,20 @@
  */
 import React, { useEffect } from 'react';
 import { useHistory } from 'react-router-dom';
-import { fetchHasData } from '../../data_handler';
-import { useFetcher } from '../../hooks/use_fetcher';
-import { useQueryParams } from '../../hooks/use_query_params';
+import { useHasData } from '../../hooks/use_has_data';
 import { LoadingObservability } from '../overview/loading_observability';
 
 export function HomePage() {
   const history = useHistory();
-
-  const { absStart, absEnd } = useQueryParams();
-
-  const { data = {} } = useFetcher(
-    () => fetchHasData({ start: absStart, end: absEnd }),
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    []
-  );
-
-  const values = Object.values(data);
-  const hasSomeData = values.length ? values.some((hasData) => hasData) : null;
+  const { hasAnyData } = useHasData();
 
   useEffect(() => {
-    if (hasSomeData === true) {
+    if (hasAnyData === true) {
       history.push({ pathname: '/overview' });
-    }
-    if (hasSomeData === false) {
+    } else if (hasAnyData === false) {
       history.push({ pathname: '/landing' });
     }
-  }, [hasSomeData, history]);
+  }, [hasAnyData, history]);
```

Comment:
I think this would be more understandable if we use `hasAnyData: boolean` and `isAllLoaded:boolean`. I'm not a big fan of using `boolean | undefined`.